astropy/units/tests/test_quantity.py

```diff
@@ -1055,12 +1055,31 @@ def test_quantity_pickelability():
     assert q1.unit == q2.unit
 
 
-def test_quantity_from_string():
-    with pytest.raises(TypeError):
-        q = u.Quantity(u.m * "5")
-        # the reverse should also fail once #1408 is in
+def test_quantity_initialisation_from_unit():
+    q = u.Quantity(u.m)
+    assert q.unit == u.m
+    assert q.value == 1.
+    assert u.Unit(q) == u.m
+    q = u.Quantity('m')
+    assert q.unit == u.m
+    assert q.value == 1.
+    assert u.Unit(q) == u.m
+    q = u.Quantity('')
+    assert q.unit == u.dimensionless_unscaled
+    assert q.value == 1.
+    assert u.Unit(q) == u.dimensionless_unscaled
+    q = u.Quantity(u.m * "5")
+    assert q.unit == u.m  # not "is", since unit is CompositeUnit
+    assert q.value == 5.
+    assert u.Unit(q) == u.Unit(u.m * "5")
+    q = u.Quantity('1.5 m/s')
+    assert q.unit == u.m/u.s
+    assert q.value == 1.5
+    assert u.Unit(q) == u.Unit('1.5 m/s')
 
-    with pytest.raises(TypeError):
+
+def test_quantity_string_value_invalid():
+    with pytest.raises(u.UnitsError):
         q = u.Quantity('5', u.m)
```

Comment:
I would expect this to work since `'5 m'` works and `'5' * u.m` works.  And the `UnitsError` seems especially confusing since there is nothing wrong with the units.